[inductor] Remove flag of bmm's dim m and n in shape padding

jiawenliu64 opened the pull request:
Summary: There was an OOM issue in two internal models when turning on padding bmm with dim m and n with shape padding optimization, so added a flag to turned on/off for the internal models. The issue was gone now so removing the flag.

Differential Revision: D42074557



cc @mlazos @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @chunyuan-w @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

ngimel commented on torch/_inductor/decomposition.py:
```diff
@@ -361,7 +357,7 @@ def pad_bmm(mat1, mat2, m_padded_length, k_padded_length, n_padded_length):
         mat2 = pad_dim(mat2, k_padded_length, 1)
         return torch.ops.aten.bmm(mat1, mat2)
-    elif config.shape_padding_bmm and n_padded_length != 0:
+    elif n_padded_length != 0:
```
Can you leave a comment here that you go through decomp multiple times if you have to pad multiple dimensions? `elif`s are a bit confusing

jiawenliu64 replied:
Comment added

Commit:
[inductor] Remove flag of bmm's dim m and n in shape padding (#90937)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/90937

There was an OOM issue in two internal models when turning on padding bmm with dim m and n with shape padding optimization, so added a flag to turned on/off for the internal models. The issue was gone now so removing the flag.

Differential Revision: D42074557

fbshipit-source-id: bcf80a691f8184b647773632a60577f9bc9728b5
